solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListView.swift

```diff
@@ -3,95 +3,106 @@
-  private var listItems: [RepositoryCellViewConfiguration] = []
-
-  private lazy var tableView: UITableView = {
-    let tableView = UITableView(frame: .zero)
-    tableView.translatesAutoresizingMaskIntoConstraints = false
-    tableView.register(
-      RepositoryCellView.self,
-      forCellReuseIdentifier: RepositoryCellView.reuseIdentifier
-    )
-    tableView.dataSource = self
-    tableView.delegate = self
-    tableView.rowHeight = UITableView.automaticDimension
-    tableView.estimatedRowHeight = 70
-    return tableView
-  }()
-
-  init() {
-    super.init(frame: .zero)
-
-    self.customizeInterface()
-  }
-
-  required init?(coder _: NSCoder) {
-    fatalError("init(coder:) has not been implemented")
-  }
+    // MARK: Properties
+    private var listItems: [RepositoryCellViewConfiguration] = []
+    
+    var didSelectRow: ((RepositoryCellViewConfiguration) -> Void)?
+    
+    // MARK: Outlets
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.register(
+            RepositoryCellView.self,
+            forCellReuseIdentifier: RepositoryCellView.reuseIdentifier
+        )
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.estimatedRowHeight = 70
+        return tableView
+    }()
+    
+    // MARK: Initialization
+    init() {
+        super.init(frame: .zero)
+        
+        self.customizeInterface()
+    }
+    
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
 }
 
+// MARK: Extensions
 extension ListView {
-  private func customizeInterface() {
-    self.backgroundColor = .white
-    self.configureSubviews()
-    self.configureSubviewsConstraints()
-  }
-
-  private func configureSubviews() {
-    self.addSubview(self.tableView)
-  }
-
-  private func configureSubviewsConstraints() {
-    NSLayoutConstraint.activate([
-      self.tableView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
-      self.tableView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
-      self.tableView.topAnchor.constraint(equalTo: self.topAnchor),
-      self.tableView.bottomAnchor.constraint(equalTo: self.bottomAnchor),
-    ])
-  }
+    private func customizeInterface() {
+        self.backgroundColor = .white
+        self.configureSubviews()
+        self.configureSubviewsConstraints()
+    }
+    
+    private func configureSubviews() {
+        self.addSubview(self.tableView)
+    }
+    
+    private func configureSubviewsConstraints() {
+        NSLayoutConstraint.activate([
+            self.tableView.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            self.tableView.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            self.tableView.topAnchor.constraint(equalTo: self.topAnchor),
```

Comment:
change with the to `safeAreaLayoutGuide.top`
```suggestion
            self.tableView.topAnchor.constraint(equalTo: self.safeAreaLayoutGuide.topAnchor),
```